### PR TITLE
Remove ram_file usage

### LINKED ISSUE #99: 'ram_file:uncompress' was removed in erlang/otp master
See [e95f648](https://github.com/erlang/otp/commit/e95f648)

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #99

Based on https://github.com/erlang/otp/pull/2802/files#diff-01a26a4de54e29a7b2de0a3571c3ddddaec0d9555e3ae3cf39eb2ebbd8498d99

### CODE REVIEW

**Review by ericmj — APPROVED** [review]
Easier fix than I thought :). 💜

**Comment by wojtekmach** [comment]
@ericmj i made some additional changes, could you take a second look?

File: src/hex_tarball.erl [hunk 1]
```diff
@@ -394,7 +395,12 @@ guess_build_tools(Metadata) ->
 %%====================================================================
 
 unpack_tarball(ContentsBinary, memory) ->
-    hex_erl_tar:extract({binary, ContentsBinary}, [memory, compressed]);
+    try hex_erl_tar:extract({binary, ContentsBinary}, [memory, compressed]) of
+      Ok -> Ok
+    catch
+      error:data_error ->
```
Comment (ericmj) [inline src/hex_tarball.erl:401]: Where does this error come from?

Comment (wojtekmach) [reply src/hex_tarball.erl:401]:
```
iex> :zlib.gunzip("badgzip")
** (ErlangError) Erlang error: :data_error
    :zlib.inflate_nif(#Reference<0.540589784.484573189.118669>, 8192, 16384, 0)
    :zlib.dequeue_all_chunks_1/3
    :zlib.inflate/3
    :zlib.gunzip/1
```

one of the existing tests caught it. I thought about not handling it but I thought we should.

Comment (wojtekmach) [reply src/hex_tarball.erl:401]: just to be clear, to get this error someone would need to have a correct outer tarball but incorrectly gzipped inner tarball, so yeah, pretty uncommon.

Comment (wojtekmach) [reply src/hex_tarball.erl:401]: hmm, it's a change in behaviour:

on 23.1.1:

```
~% asdf shell erlang 23.1.1 ; erl -noshell -eval 'io:format("~p~n", [erl_tar:extract({binary, <<"bad">>}, [compressed])]), halt().'
{error,eof}
```

on master:

```
~% asdf shell erlang git ; erl -noshell -eval 'io:format("~p~n", [erl_tar:extract({binary, <<"bad">>}, [compressed])]), halt().'
{"init terminating in do_boot",{data_error,[{zlib,inflate_nif,[#Ref<0.1770301242.1831469062.81859>,8192,16384,0],[]},{zlib,dequeue_all_chunks_1,3,[]},{zlib,inflate,3,[]},{zlib,gunzip,1,[]},{erl_tar,open1,4,[{file,"erl_tar.erl"},{line,326}]},{erl_tar,foldl_read,4,[{file,"erl_tar.erl"},{line,1382}]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,232}]}]}}
init terminating in do_boot ({data_error,[{zlib,inflate_nif,[,8192,16384,0],[]},{zlib,dequeue_all_chunks_1,3,[]},{zlib,inflate,3,[]},{zlib,gunzip,1,[]},{erl_tar,open1,4,[{_},{_}]},{erl_tar,foldl_read,

Crash dump is being written to: erl_crash.dump...done
```

so I think we should fix it in otp, I'll send a patch.

**Review by ericmj — APPROVED** [review]

File: src/hex_tarball.erl (hunk 1 above)
Comment (ericmj) [reply src/hex_tarball.erl:401]: Great find!

Comment (ericmj) [reply src/hex_tarball.erl:401]: Maybe add a note that the workaround is not needed after OTP 23.x?

Comment (wojtekmach) [reply src/hex_tarball.erl]: José already fixed it upstream :D https://github.com/erlang/otp/pull/2814 (I expect this to land as is, if there are any changes I'll re-apply them later.)

**Pull request merged**